src/connection/mqtt.rs

```diff
@@ -0,0 +1,619 @@
+/*
+
+* This file is part of Astarte.
+*
+* Copyright 2021 SECO Mind Srl
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+use std::{collections::HashMap, fmt::Display, ops::Deref, sync::Arc};
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use log::{debug, error, info, trace, warn};
+use once_cell::sync::OnceCell;
+use rumqttc::{Event as MqttEvent, Packet};
+use tokio::sync::Mutex;
+
+#[cfg(test)]
+pub(crate) use crate::mock::{MockAsyncClient as AsyncClient, MockEventLoop as EventLoop};
+#[cfg(not(test))]
+pub(crate) use rumqttc::{AsyncClient, EventLoop};
+
+use crate::{
+    interface::{mapping::path::MappingPath, Ownership},
+    interfaces::{Interfaces, MappingRef, ObjectRef},
+    payload::{self, Payload},
+    properties,
+    retry::DelayedPoll,
+    shared::SharedDevice,
+    store::{PropertyStore, StoredProp},
+    topic::parse_topic,
+    types::AstarteType,
+    Interface, Timestamp,
+};
+
+use super::{Connection, ReceivedEvent, Registry};
+
+pub struct SharedMqtt {
+    realm: String,
+    device_id: String,
+    eventloop: Mutex<EventLoop>,
+}
+
+pub struct Mqtt {
+    shared: Arc<SharedMqtt>,
+    client: AsyncClient,
+}
+
+impl Deref for Mqtt {
+    type Target = SharedMqtt;
+
+    fn deref(&self) -> &Self::Target {
+        &self.shared
+    }
+}
+
+impl Clone for Mqtt {
+    fn clone(&self) -> Self {
+        Self {
+            shared: Arc::clone(&self.shared),
+            client: self.client.clone(),
+        }
+    }
+}
+
+struct Introspection {
+    interfaces: String,
+    server_interfaces: Vec<String>,
+    device_properties: Vec<StoredProp>,
+}
+
+impl Introspection {
+    fn filter_device_properties(
+        properties: Vec<StoredProp>,
+        interfaces: &Interfaces,
+    ) -> Vec<StoredProp> {
+        properties
+            .into_iter()
+            .filter(|prop| {
+                matches!(interfaces.get_property(&prop.interface),
+                Some(interface) if interface.ownership() == Ownership::Device
+                    && interface.version_major() == prop.interface_major)
```

Comment:
After https://github.com/astarte-platform/astarte-device-sdk-rust/pull/234 was merged, this function is no longer needed and we can use `PropertyAccess::device_props`